daemon/Daemon.vala

```diff
@@ -200,28 +191,16 @@ namespace MayaDaemon {
             secondary_text = start_time.format ("%s, %s".printf (Granite.DateTime.get_default_date_format (false, true, true), text));
         }
 
-        if (notification == null) {
-            notification = new Notify.Notification (primary_text, secondary_text, "");
-        } else {
-            notification.clear_hints ();
-            notification.clear_actions ();
-            notification.update (primary_text, secondary_text, "");
-        }
+        var notification = new GLib.Notification (primary_text);
+        notification.set_body (secondary_text);
 
         if (missed == false) {
-            notification.icon_name = "appointment-soon";
+            notification.set_icon (new ThemedIcon ("appointment-soon"));
         } else {
-            notification.icon_name = "appointment-missed";
+            notification.set_icon (new ThemedIcon ("appointment-missed"));
         }
 
-        notification.set_urgency (Notify.Urgency.NORMAL);
-
-        try {
-            notification.show ();
-        } catch (GLib.Error err) {
-            warning ("Could not show notification: %s", err.message);
-        }
-#endif
+        GLib.Application.get_default ().send_notification (uid, notification);
```

Comment:
This does not work because Daemon.vala is not a GLib.Application (at the moment).